### DISCUSSION TITLE
Logging/Performing Logic During Retries @RetryableTopic

### DISCUSSION THREAD

**Discussion by tuckeremulls** [spring-projects/spring-kafka#2130] [Q&A] [answered]
I am using `@RetryableTopic` for a project and have a use case to log/perform actions during the retry process. Ideally, I am looking for some sort of entry point where after a record fails a retry, the amount of time before the record is sent to the next retry topic can be logged and other actions can be performed with the record's information. 

It seems like answers from [this post](https://github.com/spring-projects/spring-kafka/discussions/1914) are along the right track, however I just wanted to reach out for suggestions in case there is an ideal solution for this scenario. 

Thanks!

**Comment by garyrussell** [comment, marked as answer]
>the amount of time before the record is sent to the next retry topic

The record is sent to the next topic immediately; the delay is implemented by the consumer from that topic.

Does the `ProducerInterceptor` solution not work for some reason? If you use the default retry topic naming strategy, you can infer the retry delay from the topic name.

**Reply by tuckeremulls** [reply]
The ProducerInterceptor ended up being a perfect solution, thanks for the help!
